src/scenario.cpp

```diff
@@ -654,19 +655,15 @@ ret_val<void> scenario::can_afford( const scenario &current_scenario, const int
 ret_val<void> scenario::can_pick() const
 {
     // if meta progression is disabled then skip this
-    if( get_past_games().achievement( achievement_achievement_arcade_mode ) ||
+    if( get_past_achievements().is_completed( achievement_achievement_arcade_mode ) ||
         !get_option<bool>( "META_PROGRESS" ) ) {
         return ret_val<void>::make_success();
     }
 
     if( _requirement ) {
-        const achievement_completion_info *other_games = get_past_games().achievement(
```

Comment:
I think if delete this part, the achievements in memorial folder can't be record correctly. Maybe we also need to check  the achievements in `const past_games_info &get_past_games()`

Reply:
As far as I know this function past_games_info::achievement() just looks for the achievement in the already parsed memorial files and doesn't do any recording. This is just a check to see if its in there

Reply:
yes, the past achievement are all stored in memorial files, when we update game version and move the memorial folder to new game folder, the archievement before this PR will not be checked, so I mean that

Reply:
That is true, those memorial achievements would not get migrated over without some specific work

Reply:
That sounds good. In fact, if cdda a commercial game, make changes like clear players' even sunk cost, it 's definitely unacceptable, and players will sue the game designer.

Reply:
Memorial achievements are now being handled